rln/Cargo.toml

```diff
@@ -6,18 +6,21 @@ edition = "2021"
 [lib]
 crate-type = ["cdylib", "rlib", "staticlib"]
 
+
 [dependencies]
 
 # ZKP Generation
-ark-ec = { version = "0.3.0", default-features = false, features = ["parallel"] }
-ark-ff = { version = "0.3.0", default-features = false, features = ["parallel", "asm"] }
-ark-std = { version = "0.3.0", default-features = false, features = ["parallel"] }
+ark-ec = { version = "0.3.0", default-features = false }
+ark-ff = { version = "0.3.0", default-features = false, features = [ "asm"] }
+ark-std = { version = "0.3.0", default-features = false }
 ark-bn254 = { version = "0.3.0" }
-ark-groth16 = { git = "https://github.com/arkworks-rs/groth16", rev = "765817f", features = ["parallel"] }
+ark-groth16 = { git = "https://github.com/arkworks-rs/groth16", rev = "765817f", default-features = false }
 ark-relations = { version = "0.3.0", default-features = false, features = [ "std" ] }
 ark-serialize = { version = "0.3.0", default-features = false }
-ark-circom = { git = "https://github.com/gakonst/ark-circom", features = ["circom-2"] }
-wasmer = { version = "2.0" }
+ark-circom = { git = "https://github.com/vacp2p/ark-circom", branch = "wasm", default-features = false, features = ["circom-2"] }
```

Comment:
Can we upstream this as a PR to ark-circom too? Are all things in https://github.com/gakonst/ark-circom/issues/4 supported?

I think even if just partial support for our case it'd be a welcome PR!

Reply:
Ah, I see in https://github.com/vacp2p/ark-circom/commit/0e587145cb05e08b2d1a01509eb578670088eb2f that it requires disabling `parallels`. I wonder if it is possible to get the feature flag to be disabled when `wasm` flag is enabled? Perhaps in a conditional package list or something, WDYT?

Also any idea why `parallels` doesn't work?

Reply:
> any idea why `parallels` doesn't work?

While running with that feature enabled, the following error was generated:
```
The global thread pool has not been initialized.: ThreadPoolBuildError { kind: IOError(Error { kind: Unsupported, message: "operation not supported on this platform"
```

I was looking at https://github.com/GoogleChromeLabs/wasm-bindgen-rayon , which should enable concurrency using js web workers, but I feel like the effort of adding multithreading to zerokit-wasm should be done in a separate PR.